### PR TITLE
ALBS-656: Added new match methods in beholder client

### PR DESCRIPTION
**Pull request by palyla**
Hello everyone,
This is related to https://cloudlinux.atlassian.net/browse/ALBS-656 
Please check if I've done it properly.

### CODE REVIEW

File: alws/utils/beholder_client.py [hunk 1]
```diff
@@ -66,6 +66,33 @@ async def iter_endpoints(
                     "trying next reference platform"
                 )
 
+    async def __get_response_priority(self, response: dict, platforms: list) -> int:
```
Comment (anfimovdm) [inline alws/utils/beholder_client.py:69]: why are you using private method?

File: alws/utils/beholder_client.py [hunk 2]
```diff
@@ -66,6 +66,33 @@ async def iter_endpoints(
...
+        )
+        # We have priority only in ref platforms
+        # If the platform was not found, derive priority from the match level
+        if platform_priority is None:
+            match = response.get("match")
+            if match in {"exact", "closest"}:
+                platform_priority = ReleasePackageTrustness.MAXIMUM.value
+            elif match in {"name_version", "name_only"}:
+                platform_priority = ReleasePackageTrustness.MEDIUM.value
+            else:
```
Comment (anfimovdm) [inline alws/utils/beholder_client.py:91]: unnecessary else statement
```python3
platform_priority = LOWEST_PRIORITY
if match in {"exact", "closest"}:
    platform_priority = ReleasePackageTrustness.MAXIMUM.value
...
```

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: alws/utils/beholder_client.py (hunk 1 above)
Comment (palyla) [reply alws/utils/beholder_client.py:69]: Made it protected.

**Comment by palyla** [comment]
Screenshot 2023-07-11 at 11 01 39 AM
Screenshot 2023-07-11 at 11 01 51 AM
Screenshot 2023-07-11 at 11 01 57 AM
screencapture-192-168-246-168-release-create-2023-07-11-11_00_10
screencapture-192-168-246-168-release-create-2023-07-11-10_59_41

**Comment by palyla** [comment]
https://cloudlinux.atlassian.net/browse/ALBS-656

**Review by zklevsha — APPROVED** [review]

File: alws/release_planner.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
     PackageNevra,
     ReleasePackageTrustness,
     ReleaseStatus,
-    RepoType,
+    RepoType, BeholderMatchMethod,
```
Comment (javihernandez) [inline alws/release_planner.py:26]: Please add this one into a new line

File: alws/constants.py [hunk 4]
```diff
@@ -55,13 +56,45 @@ class PermissionTriad:
 
 
 class ReleasePackageTrustness(enum.IntEnum):
+    """
+    Enum representing the trustworthiness of a release package. The trustworthiness is shown in different colors
```
Comment (javihernandez) [inline alws/constants.py:60]: Too long lines. Rephrase or split them into several lines.

File: alws/constants.py [hunk 5]
```diff
@@ -55,13 +56,45 @@ class PermissionTriad:
...
+    MEDIUM (2): The package has medium trustworthiness, represented in yellow on the UI.
+    LOWEST (10): The package has the lowest trustworthiness, represented in red on the UI.
+
+    Methods
+    -------
+    decrease(trustness: int) -> int:
+        Decreases the level of trustness by one step from its current value, unless it's already at the lowest.
+    """
+
     UNKNOWN = 0
```
Comment (javihernandez) [inline alws/constants.py:74]: I don't get why we need `MEDIUM` to be 2 and `LOWEST` 10. If we don't plan to set the trustness to something between 2 and 10, why not doing a ```UNKNOWN, MAXIMUM, MEDIUM, LOWEST = range(4)``` here?

Comment (palyla) [reply alws/constants.py:74]: Fair enough, fixed.

**Review by Korulag — APPROVED** [review]

**Pull request merged**